lib/fog/aws/requests/compute/request_spot_instances.rb

```diff
@@ -71,7 +71,7 @@ def request_spot_instances(image_id, instance_type, spot_price, options = {})
             options.merge!(Fog::AWS.indexed_param('LaunchSpecification.SecurityGroupId', [*security_group_ids]))
           end
           if options['LaunchSpecification.UserData']
-            options['LaunchSpecification.UserData'] = Base64.encode64(options['LaunchSpecification.UserData'])
+            options['LaunchSpecification.UserData'] = Base64.encode64(options['LaunchSpecification.UserData']).chomp!
```

Comment:
It is.  the API expects the output of `strict_encode64`, but this is how this happens everywhere else, I assume for 1.8 compatibility.